bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -614,18 +613,44 @@ private byte[] macBytesFrom(InetSocketAddress inetSocketAddress) throws SocketEx
 
     /**
      * Use the given format specifier to format an array of mac address bytes
-     * 
+     *
      * @param format a standard format specifier; optionally ends with a delimiter e.g. "%02x:" or "%02X"
      * @param bytes the mac address as an array of bytes
-     * 
+     *
      * @return e.g. '01:02:03:04:A5:B6:C7:D8', '01-02-03-04:a5:b6:c7:d8', or '01020304A5B6C7D8'
      */
     private String macFormat(String format, byte[] bytes) {
         StringBuilder result = new StringBuilder();
         for (byte byt : bytes) {
             result.append(String.format(format, byt));
         }
-        boolean isDelimited = Set.of(':', '-', '.').contains(format.charAt(format.length() - 1));
+        boolean isDelimited = !Character.isLetterOrDigit(format.charAt(format.length() - 1));
         return (isDelimited ? result.substring(0, result.length() - 1) : result).toString();
     }
+
+    /**
+     * Check if the given mac format specifier string is valid
+     */
+    private boolean macFormatValid(String format) {
+        try {
+            String.format(format, (byte) 123);
+        } catch (IllegalFormatException e) {
+            return false;
+        }
+        int last = format.length() - 1;
+        int index = 0;
+        while (index <= last) {
+            if (Character.isLetter(format.charAt(index))) {
+                break;
+            }
+            index++;
+        }
+        switch (last - index) {
+            case 0:
+                return true;
+            case 1:
+                return !Character.isLetterOrDigit(format.charAt(last));
+        }
+        return false;
```

Comment:
Could you pls add a comment what this part of the method is trying to achieve?

Actually, it took me some time to get what you are trying to do (hopefully I got it correctly): 
* When there is no letter in the format string -> index will be last +1, switch will not hit a case -> false
* When there exactly one letter as last char of the format string -> the loop breaks at index=last -> switch case 0, true
* else if the letter is the 2nd last character -> break, switch will hit case "1" -> true if last char is a special character (!=letter/digit)
* else false....

seems to be the right thing... a regex-like description could be "[number]*[letter][!letterdigit]?" (this is not a proposal to use regex, comment would be sufficient)